app/controllers/application_controller.rb

```diff
@@ -450,6 +450,12 @@ def redirect_back_or_default(default, escape = true, use_escaped = true)
       begin
         uri = URI.parse(escaped_back_url)
 
+        # Remove basic auth credentials
+        uri.userinfo = nil
```

Comment:
The idea stems from the original fix, but one possible attack vector in our case would be guessing API authentication (global user credentials). 

Do we have a legitimate use case for basic auth on the application itself? It may be possible that someone uses, e.g., basic auth around an OP page, but then again the basic auth is cached by the browser itself, isn't it?